azext_iot/central/services/auth.py

```diff
@@ -0,0 +1,27 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+import base64
+import hmac
+import hashlib
+
+
+def generate_device_key(masterkey, device_id):
```

Comment:
I'd suggest maybe creating a new folder under central like /offline or something and placing this there instead